includes/User.php

```diff
@@ -4076,24 +4076,18 @@ public function incEditCount() {
 	 *
 	 * @param $password String Plain-text password
 	 * @param bool|string $salt Optional salt, may be random or the user ID.
-
 	 *                     If unspecified or false, will generate one automatically
 	 * @return String Password hash
 	 * @deprecated use HeliosClient
 	 */
 	public static function crypt( $password, $salt = false ) {
-		global $wgPasswordSalt;
+		global $wgPasswordSalt, $wgDevelEnvironment;
 		// Wikia change - begin
-		// @see PLATFORM-2502 comparing new passwords in PHP code.
-		// @todo mech remove after the new password hashing is implemented (PLATFORM-2530).
-		WikiaLogger::instance()->debug(
-			'NEW_HASHING crypt called in PHP',
-			[
-				'wgPasswordSalt' => $wgPasswordSalt,
-				'caller' => wfGetCaller(),
-				'exception' => new Exception()
-			]
-		);
+		// Only allow on devboxes for now
+		// @todo Remove once old login code is removed (PLATFORM-2891)
+		if ( empty( $wgDevelEnvironment ) ) {
+			throw new Exception( 'Unsupported User::crypt method requested' );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS112)